internal/slicer/report.go

```diff
@@ -30,37 +32,38 @@ type Report struct {
 
 // NewReport returns an empty report for content that will be based at the
 // provided root path.
-func NewReport(root string) *Report {
-	return &Report{
+func NewReport(root string) (*Report, error) {
+	if !filepath.IsAbs(root) {
+		return nil, fmt.Errorf("cannot use relative path for report root: %q", root)
+	}
+	report := &Report{
 		Root:    filepath.Clean(root) + "/",
 		Entries: make(map[string]ReportEntry),
 	}
+	return report, nil
 }
 
 func (r *Report) Add(slice *setup.Slice, fsEntry *fsutil.Entry) error {
-	if !strings.HasPrefix(fsEntry.Path, r.Root) {
-		return fmt.Errorf("cannot add path %q outside of root %q", fsEntry.Path, r.Root)
-	}
-	relPath := filepath.Clean("/" + strings.TrimPrefix(fsEntry.Path, r.Root))
-	if fsEntry.Mode.IsDir() {
-		relPath = relPath + "/"
+	path, err := r.sanitizePath(fsEntry.Path, fsEntry.Mode.IsDir())
+	if err != nil {
+		return fmt.Errorf("cannot add path: %s", err)
 	}
 
-	if entry, ok := r.Entries[relPath]; ok {
+	if entry, ok := r.Entries[path]; ok {
```

Comment:
The logic here used a relative paths and absolute paths intentfully, for security but also clarity on the user interface.  Note how we only reported the out-of-root issue independently, but then all the details below are inside the root path, and thus nicely short and clear. The variables were also aptly named to reduce the chances of mistakes while using them. All of this was careful design, and it doesn't look like the motivation for this PR requires touching these details.

Reply:
Are you suggesting changing the name back to `relPath` that is indeed more informative?

If you are talking about the refactor into `sanitizePath`, I think having it as a separate function has value because we want the logic to be the same across all entry points to the report (`Add` and `Mutate`), this way we can change the logic in one place with no possibility for drift.

Maybe what we need to do is change the name of `sanitizePath` to something more informative.